libcst/matchers/_visitors.py

```diff
@@ -75,20 +82,9 @@ def _get_valid_leave_annotations_for_classes(
         # all possible valid return annotations. Its not really possible for us (or
         # pyre) to fully enforce return types given the presence of OneOf/AllOf matchers, so
         # we do the best we can by taking a union of all valid return annotations.
-
-        # TODO: We could possibly teach LibCST codegen to generate a mapping of class
-        # to valid leave annotations when it generates leave_<Node> methods, but for
-        # now this is functionally identical. That would get rid of the need to provide
-        # a namespace here, as well as a gross getattr on the CSTTransformer class.
-        meth = getattr(cst.CSTTransformer, f"leave_{cls.__name__}")
-        namespace: Dict[str, object] = {
-            **{x: getattr(cst, x) for x in dir(cst)},
-            "cst": cst,
-            "libcst": cst,
-        }
-        type_hints = get_type_hints(meth, namespace, namespace)
-        if "return" in type_hints:
-            retval.update(_get_possible_annotated_classes(type_hints["return"]))
+        retval.update(
+            _get_possible_annotated_classes(TYPED_FUNCTION_RETURN_MAPPING[cls])
+        )
```

Comment:
Will this change the result of retval? Or it's just a refactoring.

Reply:
It is a pure refactor! It changes absolutely nothing (I verified with all tests that it is identical).